### PR TITLE
 Convert ingress provided URLs to https

### PR DESCRIPTION
**Pull request by nsklikas**
- Convert ingress provided URLs to https and remove the port.
- Remove `external_url` config.

### CODE REVIEW

File: src/utils.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+# See LICENSE file for licensing details.
+
+"""Utility functions for the Kratos charm."""
+
+from typing import Dict
+from urllib.parse import urlparse, urlunparse
+
+
+def dict_to_action_output(d: Dict) -> Dict:
+    """Convert all keys in a dict to the format of a juju action output."""
```
Comment (gruyaume) [inline src/utils.py:12]: Can you please add an example in the docstring to make it clearer what this method does?

File: src/utils.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    for k, v in d.items():
+        k = k.replace("_", "-")
+        if isinstance(v, dict):
+            v = dict_to_action_output(v)
+        ret[k] = v
+    return ret
+
+
+def normalise_url(url: str) -> str:
+    """Convert a URL to a more userfriendly format.
```
Comment (gruyaume) [inline src/utils.py:23]: Can you please add an example in the docstring to make it clearer what this method does?

File: src/utils.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Utility functions for the Kratos charm."""
+
+from typing import Dict
+from urllib.parse import urlparse, urlunparse
+
+
+def dict_to_action_output(d: Dict) -> Dict:
```
Comment (gruyaume) [inline src/utils.py:11]: You should have unit tests specifically for those new methods.

Comment (nsklikas) [reply src/utils.py:11]: Done

File: src/utils.py (hunk 2 above)
Comment (nsklikas) [reply src/utils.py]: Done

File: src/utils.py (hunk 1 above)
Comment (nsklikas) [reply src/utils.py]: Done

**Pull request merged**